### PR TITLE
[chore] Target Allocator example

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Add example showing how to use the Operator Target Allocator alongside the OpenTelemetry Collector daemonset

### CODE REVIEW

File: examples/target-allocator/target-allocator-values.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+          collector_id: ${env:K8S_POD_NAME}
+    service:
+      pipelines:
+        metrics:
+          receivers:
+            - hostmetrics
+            - kubeletstats
+            - otlp
+            - signalfx
+            - prometheus/crd # Adding this receiver to the list of metrics pipeline receivers.
```
Comment (jihuiyang-x) [inline examples/target-allocator/target-allocator-values.yaml:25]: This looks good. did you deploy this to a cluster with prometheus crd and verify it worked?

File: examples/target-allocator/target-allocator.yaml [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      app: targetallocator
+  template:
+    metadata:
+      labels:
+        app: targetallocator
+    spec:
+      serviceAccountName: opentelemetry-targetallocator
+      containers:
+        - name: targetallocator
+          image: ghcr.io/open-telemetry/opentelemetry-operator/target-allocator:v0.105.0
```
Comment (jihuiyang-x) [inline examples/target-allocator/target-allocator.yaml:88]: you are pulling the open source target allocator, is the plan to have a forked splunk-target-allocator in the future?

File: examples/target-allocator/target-allocator-values.yaml (hunk 1 above)
Comment (atoulme) [reply examples/target-allocator/target-allocator-values.yaml:25]: I did.

File: examples/target-allocator/target-allocator.yaml (hunk 2 above)
Comment (atoulme) [reply examples/target-allocator/target-allocator.yaml:88]: I pull the open source target allocator, and the plan is to use it moving forward. We will contribute upstream as needed.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**